test/e2eclusterctl/config/azure-dev.yaml

```diff
@@ -0,0 +1,60 @@
+managementClusterName: kind-capz
+
+images:
+  - name: ${MANAGER_IMAGE}
+    loadBehavior: mustLoad
+
+providers:
+  - name: cluster-api
+    type: CoreProvider
+    versions:
+      - name: v0.3.6
+        value: https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.3.6/core-components.yaml
+        type: url
+        replacements:
+          - old: "imagePullPolicy: Always"
+            new: "imagePullPolicy: IfNotPresent"
+          - old: "--enable-leader-election"
+            new: "--enable-leader-election=false"
+  - name: kubeadm
+    type: BootstrapProvider
+    versions:
+      - name: v0.3.6
+        value: https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.3.6/bootstrap-components.yaml
+        type: url
+        replacements:
+          - old: "imagePullPolicy: Always"
+            new: "imagePullPolicy: IfNotPresent"
+          - old: "--enable-leader-election"
+            new: "--enable-leader-election=false"
+  - name: kubeadm
+    type: ControlPlaneProvider
+    versions:
+      - name: v0.3.6
+        value: https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.3.6/control-plane-components.yaml
+        type: url
+        replacements:
+          - old: "imagePullPolicy: Always"
+            new: "imagePullPolicy: IfNotPresent"
+          - old: "--enable-leader-election"
+            new: "--enable-leader-election=false"
+  - name: azure
+    type: InfrastructureProvider
+    versions:
+      - name: v0.4.4
```

Comment:
How will this be maintained in the future? @fabriziopandini do we need to update the release every time after cutting a new release?

Reply:
It looks like we only have to maintain the major and minor version of this field, and they have to match whatever is defined in https://github.com/kubernetes-sigs/cluster-api/blob/8b29a9adeeba4677da4221b887cce61155f925a3/cmd/clusterctl/client/repository/metadata_client.go#L189. I can set it to `v0.4.0` (following the convention of [capi](https://github.com/kubernetes-sigs/cluster-api/blob/master/test/e2e/config/docker-dev.yaml#L79 ) and [capa](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1716/files#diff-210cac49a90b04e3af86b95cc1169327R76)) so we don't have to change it unless we are cutting a new minor release.